common/djangoapps/student/tests/test_models.py

```diff
@@ -585,6 +585,17 @@ def test_retiring_nonexistent_user_doesnt_modify_records(self):
         )
         assert user_search_results.exists()
 
+    def test_former_allowed_student_isnt_counted_if_email_is_updated(self):
+        """
+        Make sure that if allowed students change their emails after being registered, then they are not counted.
+        """
+        assert 0 == CourseEnrollmentAllowed.may_enroll_and_unenrolled(course_id=self.course_key).count()
+
+        CourseEnrollmentFactory.create(user=self.user, course_id=self.course_key, is_active=True)
+        self.allowed_enrollment.email = 'updated-learner@example.com'
```

Comment:
can you test if it counts users that unenroll from the course?

Reply:
Indeed, that's a great idea because the current behavior is that the unenrolled `CourseEnrollments` aren't counted. Therefore is worth it to test this case.